docs/docs/guides/labs/dg/incrementally-adopting-dg/migrating-project.md

```diff
@@ -94,7 +94,73 @@ There are three settings:
 
 Now that these settings are in place, you can interact with your project using `dg`. If we run `dg list defs` we can see the sole existing asset in our project:
 
-<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/4-list-defs.txt"  />
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/5-list-defs.txt"  />
+
+### Add a `dagster_dg.plugin` entry point
+
+We're not quite done adding configuration. `dg` uses the Python [entry
+point](https://packaging.python.org/en/latest/specifications/entry-points/) API
+to expose custom component types and other scaffoldable objects from user
+projects. Our entry point declaration will specify a submodule as the location
+where our project exposes plugin objects. By convention, this submodule is
+named `<root_module>.lib`. In our case, it will be `my_existing_project.lib`.
+Let's create this submodule now:
+
+<CliInvocationExample
+path="docs_snippets/docs_snippets/guides/dg/migrating-project/6-create-lib.txt"
+/>
+
+:::tip
+See the [plugin guide](/guides/labs/components/creating-new-component-types/creating-dg-plugin) for more on `dg` plugins.
+:::
+
+We'll need to add a `dagster_dg.plugin` entry point to our project and then
+reinstall the project package into our virtual environment. The reinstallation
+step is crucial. Python entry points are registered at package installation
+time, so if you simply add a new entry point to an existing editable-installed
+package, it won't be picked up.
+
+Entry points can be declared in either `pyproject.toml` or `setup.py`:
+
+<Tabs groupId="package-manager">
+    <TabItem value="uv" label="uv">
+        Since our package metadata is in `pyproject.toml`, we'll add the entry
+        point declaration there:
+
+        <CodeExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/7-uv-plugin-config.toml" language="toml" title="pyproject.toml" />
+
+        Then we'll reinstall the package. Note that `uv sync` will _not_
+        reinstall our package, so we'll use `uv pip install` instead:
+        <CodeExample path="docs_snippets/docs_snippets/guides/dg/migrating-project/8-uv-reinstall-package.txt" />
+        
+    </TabItem>
+    <TabItem value="pip" label="pip">
+        Our package metadata is in `setup.py`. While it is possible to add
+        entry point declarations to `setup.py` directly, we want to be able to
+        read the entry point declaration from `dg`, and there is no reliable
+        way to read `setup.py` (since it is arbitrary Python code). So we'll
+        instead add the entry point to a new `setup.cfg`, which can be used
+        alongside `setup.py`. Create `setup.cfg` with the following contents
+        (if your package has existing entry points declared in `setup.py`, you'll
+        want to move their definitions to `setup.cfg` as well):
```

Comment:
<img width="923" alt="image" src="https://github.com/user-attachments/assets/fee9e642-8ce4-4ce1-9e71-5c1e929640cd" />

Indentation of example is off